api/src/main/java/net/kyori/adventure/text/Component.java

```diff
@@ -47,7 +47,41 @@
 import org.checkerframework.checker.nullness.qual.Nullable;
 
 /**
- * A component.
+ * A component is ultimately text that can be displayed to the Minecraft client through different mediums.
+ *
+ * Some rules apply to all types of components:
+ * <ul>
+ *
+ *   <li>
+ *     Components are <i>immutable</i> after they are built/created.
+ *   <p> This means that:
+ *      <pre>
+ *      myComponent.style(style)
+ *      </pre>
+ *   will not change "myComponent", but rather return a new component with the content from and
+ *   the styles of the old component in addition to the new style. The same logic applies when
+ *   applying decorations/colors, appending other components and every other operation that
+ *   modifies components after they are created.
+ *   </li>
+ *
+ *   <li>
+ *     All components can be built using their respective builders which are <i>mutable</i>.
+ *     In addition to creating components from builders, all components can also be turned into builders.
+ *   </li>
+ *
+ *   <li>
+ *     The content of a component can not be directly fetched from the object itself, to get this you
+ *     need a <a href="https://docs.adventure.kyori.net/serializer/index.html">text serializer</a>.
+ *   </li>
+ * </ul>
+ *
+ * While there are many different types of components they all have the same way of modification when it comes to styling,
+ * what differentiates them is what they take as an input and how they use that input to display something.
+ * To understand what the different types of components can do you can either read their interfaces or you can go to the official documentation.
+ *
+ *
+ * <p> The official documentation for this library can be found <a href="https://docs.adventure.kyori.net/index.html">here</a>
```

Comment:
...is this necessary in the `Component` class? This is better suited for a `package-info.java` at the root of the API.